cms/djangoapps/xblock_config/migrations/0003_move_course_edit_lti_fields_enabled_flag_model_to_lti_consumer.py

```diff
@@ -0,0 +1,22 @@
+# Generated by Django 2.2.20 on 2021-05-05 12:03
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('xblock_config', '0002_courseeditltifieldsenabledflag'),
+    ]
+
+    operations = [
+        # Since this model only needs to be moved, we just want to
+        # indicate to Django that the model is no longer here without
+        # actually deleting any data from the database.
+        migrations.SeparateDatabaseAndState(
```

Comment:
I am not entirely sure what's happening here in this migration. Can you provide some details?

Reply:
Django migrations work on two levels. One is the internal state/representation of the models, and the other is what is applied to the database. 

For instance, when you remove a field and generate migrations, Django will go through all previous migrations and build an internal representation of what the model should be; then it will look at the current state of the model, and see that a field was removed. This will generate a `RemoveField` operation in the next migration file. This will cause Django to update its internal representation of the model, so it won't ask you to generate the migration again and again. When you run the migration it will also apply this change to the database. 

What the `SeparateDatabaseAndState` operation does is to separate the action Django performs in python and the action it performs in the database. 

Here simply deleting the model will also delete the data in the database, but we don't want that, we want to keep the data in the database, but still indicate to Django that this model is no more. So here we are telling it to delete the model from the django state, but not apply that to the database, which is why the delete operation is applied in `state_operations` but no in `database_operations`.